python/python/glide/async_commands/core.py

```diff
@@ -132,20 +132,124 @@ class InfBound(Enum):
     NEG_INF = "-inf"
 
 
-class ScoreLimit:
+class LexInfBound(Enum):
     """
-    Represents a score limit in a sorted set.
+    Enumeration representing lexicographic positive and negative infinity bounds for sorted set scores.
+    """
+
+    POS_INF = "+"
+    NEG_INF = "-"
+
+
+class ScoreBoundary:
+    """
+    Represents a score boundary in a sorted set.
 
     Args:
         value (float): The score value.
-        is_inclusive (bool): Whether the score value is inclusive. Defaults to False.
+        is_inclusive (bool): Whether the score value is inclusive. Defaults to True.
     """
 
     def __init__(self, value: float, is_inclusive: bool = True):
-        """Convert the score limit to the Redis protocol format."""
+        """Convert the score boundary to the Redis protocol format."""
         self.value = str(value) if is_inclusive else f"({value}"
 
 
+class LexBoundary:
+    """
+    Represents a lexicographic boundary in a sorted set.
+
+    Args:
+        value (str): The lex value.
+        is_inclusive (bool): Whether the score value is inclusive. Defaults to True.
+    """
+
+    def __init__(self, value: str, is_inclusive: bool = True):
+        """Convert the lexicographic boundary to the Redis protocol format."""
+        self.value = f"[{value}" if is_inclusive else f"({value}"
+
+
+class Limit:
+    """
+    Represents a limit argument for a range query in a sorted set to be used in ZRANGE command.
+
+    The optional LIMIT argument can be used to only get a range of the matching elements
```

Comment:
I think the documentation in redis is better, don't change it if there isn't a good reason:
```
The optional LIMIT argument can be used to obtain a sub-range from the matching elements